### ISSUE TITLE
Supervisor should use volume long syntax for labels

### ISSUE THREAD

**Issue by cywang117** [balena-os/balena-supervisor#1933]
Short vs long syntax: https://docs.docker.com/compose/compose-file/compose-file-v2/#volumes

Currently, balena-compose-parse supports version 2.4, but everywhere else still does not. Per https://github.com/balena-os/balena-supervisor/pull/1780, we want to move to using Mounts instead of Binds, to avoid instances of https://github.com/balena-os/balena-supervisor/issues/1231. In [addFeaturesFromLabels](https://github.com/balena-os/balena-supervisor/blob/104a8006fbab4bad0f2da923eade0e64002dd59f/src/compose/utils.ts#L321), we add volumes derived from balena's docker-compose labels with the short syntax only. We should use the long syntax here instead, since some labels involve bind mounts to a socket. This will require adding support for long syntax.

This will allow us to distinguish between labels that should use the long syntax (Mounts API), and internal bind mounts that should stay with using the short syntax. The latter is namely for update locks:
```
/tmp/balena-supervisor/services/$APP_ID/$SERVICE_NAME:/tmp/balena
/tmp/balena-supervisor/services/$APP_ID/$SERVICE_NAME:/tmp/resin
```
which should remain as the short syntax, since bind mounts using HostConfig.Binds config auto-creates the directory on host if it does not exist, and /tmp directories will not exist following device reboots.

Flowdock: https://www.flowdock.com/app/rulemotion/r-supervisor/threads/6b66VzeJZynWcuevc4tuvCEaOrC

**Comment by cywang117** [comment]
This issue can be broken up into two PRs, per @pipex:
- Add support for long volume syntax, where volumes being described using an object would go to Mount and those using a string would go to Bind (minor change)
- Use long volume syntax for supervisor label mounts (breaking change)

**Comment by cywang117** [comment]
Resolved with Supervisor v14.0.0.
